src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/talismans/MagicianTalisman.java

```diff
@@ -30,22 +31,31 @@
 public class MagicianTalisman extends Talisman {
 
     private final ItemSetting<Boolean> allowEnchantmentBooks = new ItemSetting<>("allow-enchantment-books", false);
+    private final ItemSetting<Boolean> allowCustomEnchantments = new ItemSetting<>("allow-custom-enchants", false);
 
     private final Set<TalismanEnchantment> enchantments = new HashSet<>();
 
     @ParametersAreNonnullByDefault
     public MagicianTalisman(SlimefunItemStack item, ItemStack[] recipe) {
         super(item, recipe, false, false, "magician", 80);
 
-        addItemSetting(allowEnchantmentBooks);
+        addItemSetting(allowEnchantmentBooks, allowCustomEnchantments);
 
         for (Enchantment enchantment : Enchantment.values()) {
-            try {
-                for (int i = 1; i <= enchantment.getMaxLevel(); i++) {
-                    enchantments.add(new TalismanEnchantment(enchantment, i));
+            /* 
+             * Stop custom enchant plugins appearing
+             * We also want to stop some stupid plugins which register as Minecraft enchants
+             * but also go above the max enchant level.
+             */
+            if ((allowCustomEnchantments.getValue() || enchantment.getKey().getNamespace().equals(NamespacedKey.MINECRAFT))) {
+                try {
+                    // Make sure we cap this at max level or if set it incorrectly, use Short.MAX_VALUE
+                    for (int i = 1; i <= Math.max(enchantment.getMaxLevel(), Short.MAX_VALUE); i++) {
```

Comment:
You probably meant `min` here :monkaS:
```suggestion
                    for (int i = 1; i <= Math.min(enchantment.getMaxLevel(), Short.MAX_VALUE); i++) {
```

Reply:
*facepalm* I swear, I do know how to code